common/src/main/java/tds/common/health/HealthIndicatorClient.java

```diff
@@ -0,0 +1,66 @@
+package tds.common.health;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Rest client to Spring Boot’s Actuator Project health check endpoint.
+ *
+ * @see <a href="http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready">Spring Boot’s Actuator Project</a>
+ */
+public class HealthIndicatorClient {
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public HealthIndicatorClient(final RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+        restTemplate.setErrorHandler(new DefaultResponseErrorHandler() {
```

Comment:
I know this works, but it feels cleaner to me to either do this before the assignment to `this.restTemplate` or call `this.restTemplate.setErrorHandler()`.